synapse/lib/storm.py

```diff
@@ -1425,12 +1426,23 @@ def parse_args(self, argv):
 
         return retn
 
+    def _norm_stormtype(self, argtype):
+        def normarg(valu):
+            norm = s_datamodel.Model().type(argtype).norm(valu)
+            return norm[0]
+        return normarg
+
     def _get_store(self, name, argdef, todo, opts):
 
         dest = argdef.get('dest')
         nargs = argdef.get('nargs')
         argtype = argdef.get('type')
 
+        if argtype in ('time', 'ival'):
```

Comment:
This feels like a step in the right direction.  maybe we should just *always* require that type is a valid model type name so we get this behavior for arguments other than time/ival?

Reply:
Updated to allow other types and require that 'type' is the name of a model type.